src/Microsoft.DotNet.Web.ProjectTemplates/content/RazorPagesWeb-CSharp/Pages/Account/ConfirmEmail.cshtml.cs

```diff
@@ -28,13 +28,13 @@ public ConfirmEmailModel(UserManager<ApplicationUser> userManager)
             var user = await _userManager.FindByIdAsync(userId);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with ID '{userId}'.");
+                return NotFound($"Unable to load user with ID '{userId}'.");
             }
 
             var result = await _userManager.ConfirmEmailAsync(user, code);
             if (!result.Succeeded)
             {
-                throw new ApplicationException($"Error confirming email for user with ID '{userId}':");
+                throw new InvalidOperationException($"Error confirming email for user with ID '{userId}':");
```

Comment:
@Eilon Should this be a new MVC `IActionResult` (I don't think any of the existing ones fit)? There are other below which have "unexpected errors" like this

Reply:
I've given this some thought and I think that for this PR this is fine. It's a definite improvement from what we had before. We're going to do a full template review at a later time to see if we wish to make any further changes to the templates in cases like this.